### PR TITLE
refactor: refactoring code around incoming receipts handling

### PR DESCRIPTION
**Pull request by wacban**
Refactoring some methods handling the incoming receipts. It's a warm up before handling that during resharding :).

### CODE REVIEW

File: chain/client/src/client.rs [hunk 1]
```diff
@@ -1506,122 +1506,134 @@ impl Client {
         }
 
         if let Some(validator_signer) = self.validator_signer.clone() {
```
Comment (wacban) [inline chain/client/src/client.rs:1508]: This was refactored to two new methods:
- reconcile_transaction_pool
- produce_chunks

**Review by wacban — COMMENTED** [review]
best reviewed with hide whitespace enabled

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -1711,17 +1711,18 @@ impl Chain {
...
-                    self.store.get_partial_chunk(&chunk_header.chunk_hash()).unwrap();
-                for receipt in partial_encoded_chunk.receipts().iter() {
-                    let ReceiptProof(_, shard_proof) = receipt;
-                    let ShardProof { from_shard_id: _, to_shard_id, proof: _ } = shard_proof;
-                    receipt_proofs_by_shard_id
-                        .entry(*to_shard_id)
-                        .or_insert_with(Vec::new)
-                        .push(receipt.clone());
-                }
+            if chunk_header.height_included() != height {
```
Comment (nikurt) [inline chain/chain/src/chain.rs:1714]: Took me a while to parse the comment before this function (lines 1698-1701).

**Review by nikurt — APPROVED** [review]
Splitting big functions into smaller functions is always welcome :+1: :smiley:

**Pull request merged**